src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -28,1625 +28,1626 @@
-                        _parent._waj._dataPager.Sync(Interlocked.Read(ref _parent._totalWrittenButUnsyncedBytes));
-                        if (_parent._waj._logger.IsInfoEnabled)
-                        {
-                            var sizeInKb = (_parent._waj._dataPager.NumberOfAllocatedPages * Constants.Storage.PageSize) / Constants.Size.Kilobyte;
-                            _parent._waj._logger.Info($"Sync of {sizeInKb:#,#0} kb file with {_currentTotalWrittenBytes / Constants.Size.Kilobyte:#,#0} kb dirty in {sp.Elapsed}");
-                        }
-                    }
-                }
+        foreach (var journalFile in unusedJournals.OrderBy(x => x.Number))
+        {
+          journalFile.FreeScratchPagesOlderThan(txw, lastSyncedTransactionId);
+        }
 
-                private bool WaitToGatherInformationToStartSync()
-                {
-                    return _parent._flushLockTaskResponsible.WaitForTaskToBeDone(GatherInformationToStartSync);
-                }
+        foreach (var jrnl in _waj._files.OrderBy(x => x.Number))
+        {
+          jrnl.FreeScratchPagesOlderThan(txw, lastSyncedTransactionId);
+        }
 
-                private void GatherInformationToStartSync()
-                {
-                    if (_parent._waj._env.Disposed)
-                        return; // we have already disposed, nothing to do here
-
-                    if (_parent._lastFlushedJournal == null)
-                        // nothing was flushed since we last synced, nothing to do
-                        return;
-
-                    _currentTotalWrittenBytes = Interlocked.Read(ref _parent._totalWrittenButUnsyncedBytes);
-                    _lastSyncedJournal = _parent._lastFlushedJournalId;
-                    _lastSyncedTransactionId = _parent._lastFlushedTransactionId;
-                    _parent._lastFlushedJournal.SetLastReadTxHeader(_lastSyncedTransactionId, ref _transactionHeader);
-                    if (_lastSyncedTransactionId != _transactionHeader.TransactionId)
-                    {
-                        VoronUnrecoverableErrorException.Raise(_parent._waj._env,
-                            $"Error syncing the data file. The last sync tx is {_lastSyncedTransactionId}, but the journal's last tx id is {_transactionHeader.TransactionId}, possible file corruption?"
-                        );
-                    }
-
-                    _parent._lastFlushedJournal = null;
-
-                    foreach (var toDelete in _parent._journalsToDelete)
-                    {
-                        if (toDelete.Key > _lastSyncedJournal)
-                            continue;
-
-                        _journalsToDelete.Add(toDelete);
-                    }
-
-                    _parent._waj._env.Options.SetLastReusedJournalCountOnSync(_journalsToDelete.Count);
-
-                    foreach (var kvp in _journalsToDelete)
-                    {
-                        _parent._journalsToDelete.Remove
```

Comment:
where are you reloading the `_task`_ instance?

Reply:
In the beginning of `WaitForTaskToBeDone` fuction [<sup>ref</sup>](https://github.com/ravendb/ravendb/pull/8188/files#diff-33af42ec0439152b32cc47c6739f3211R960) if I understand the question wright.

Reply:
No, I'm refering to loading the value on the variable, which may be cached in a register.
Talk to me about this tomorrow, please